### DISCUSSION TITLE
Pocketbase auth cached?

### DISCUSSION THREAD

**Discussion by bddylol** [pocketbase/pocketbase#4739] [Q&A]
I have a website that uses Next.js and pocketbase; when I edit the username for example on pocketbase dashboard; why is the username on my website still cached?
![image](https://github.com/pocketbase/pocketbase/assets/55510116/a8f1173d-b383-42c0-9ee7-fc7272735236)
![image](https://github.com/pocketbase/pocketbase/assets/55510116/2f100fdd-5e4f-4778-b44a-30f9ad8b1cb6)

**Comment by ganigeorgiev** [comment]
We don't cache the responses (at least not out of the box) and most likely your issue is related to the [default Next.js caching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#caching-data).

Make sure to check ["Opting out of Data Caching"](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#opting-out-of-data-caching) (when using the JS SDK additional fetch params can be passed in the [last optional `SendOptions` argument](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#custom-request-options)).

**Reply by bddylol** [reply]
Hmm, I am using the js-sdk though; I'm using this code to fetch the user:
```ts
  async getUser(cookieStore: ReadonlyRequestCookies) {
    const cookie = cookieStore.get("pb_auth");
    if (!cookie) {
      return false;
    }

    this.client.authStore.loadFromCookie(cookie?.value || "");
    return this.client.authStore.model;
  }
```

**Reply by ganigeorgiev** [reply]
I don't understand what is the expected behavior.

The above code doesn't send any request and just populate the store with whatever object is serialized in the cookie. Keep in mind that you can't trust the cookie value unless you send `authRefresh()` or other similar request to the PocketBase server.

Additionally note that PocketBase APIs are fully stateless and doesn't use cookies. The cookie is just to share the token and pb auth state between the browser and your node server.

_As a side-note, if you are new to web development I don't recommend using JS SSR with PocketBase (_Next.js is especially difficult to work with_). Some more ramblings on this you can find in https://github.com/pocketbase/pocketbase/discussions/4412#discussioncomment-8575834, https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7620032, and probably others._
